konduit-serving-pipeline/src/main/java/ai/konduit/serving/pipeline/impl/step/ml/ssd/SSDToBoundingBoxRunner.java

```diff
@@ -52,37 +56,41 @@ public Data exec(Context ctx, Data data) {
 
         String key = "detection_boxes";     //TODO
         String prob = "detection_scores";
+        String labels = "detection_classes";
 
         NDArray bND = data.getNDArray(key);
         NDArray pND = data.getNDArray(prob);
+        NDArray lND = data.getNDArray(labels);
 
         float[][][] bArr = bND.getAs(float[][][].class);        //Batch, num, xy
         float[][] pArr = pND.getAs(float[][].class);            //Batch, num
-
+        float[][] lArr = lND.getAs(float[][].class);
+        System.out.println(lND);
         List<BoundingBox> l = new ArrayList<>();
-        for(int i=0; i<bArr[0].length; i++ ){
+        for (int i = 0; i < bArr[0].length; i++) {
             //SSD order usually: [y1, x1, y2, x2]
             double y1 = bArr[0][i][0];
             double x1 = bArr[0][i][1];
             double y2 = bArr[0][i][2];
             double x2 = bArr[0][i][3];
             double p = pArr[0][i];
-            if(p >= threshold) {
-                l.add(BoundingBox.createXY(x1, x2, y1, y2, null, p));
+            float label = lArr[0][0];
+            if (p >= threshold) {
+                l.add(BoundingBox.createXY(x1, x2, y1, y2, COCO_CLASSES_LABELS[(int) label - 1], p));
```

Comment:
We should add a label index check here, and throw a helpful exception like: "Predicted label index was 30 but only 10 labels were provided"